src/pubsub/src/subscriber/builder.rs

```diff
@@ -0,0 +1,174 @@
+}
+
+impl<S> StreamingPull<S>
+where
+    S: Stub,
+{
+    pub(crate) fn new(inner: Arc<S>, subscription: String) -> Self {
+        Self {
+            inner,
+            subscription,
+            ack_deadline_seconds: 10,
+            max_outstanding_messages: 1000,
+            max_outstanding_bytes: 100 * 1024 * 1024,
+        }
+    }
+
+    /// Sets the ack deadline to use for the stream.
+    ///
+    /// This value represents how long the application has to ack or nack an
+    /// incoming message. Note that this value is independent of the deadline
+    /// configured on the server-side subscription.
+    ///
+    /// The minimum deadline you can specify is 10 seconds. The maximum deadline
+    /// you can specify is 600 seconds (10 minutes).
+    ///
+    /// The default value is 10 seconds.
+    ///
+    /// To use the server-side subscription deadline, specify a value of `0`.
+    ///
+    /// # Example
+    ///
+    /// ```no_rust
+    /// # use google_cloud_pubsub::client::Subscriber;
+    /// # async fn sample() -> anyhow::Result<()> {
+    /// # let client = Subscriber::builder().build().await?;
+    /// let session = client.streaming_pull("projects/my-project/subscriptions/my-subscription")
+    ///     .set_ack_deadline_seconds(20)
+    ///     .start();
+    /// # Ok(()) }
+    /// ```
+    pub fn set_ack_deadline_seconds<T: Into<i32>>(mut self, v: T) -> Self {
+        self.ack_deadline_seconds = v.into();
+        self
+    }
+
+    /// Flow control settings for the maximum number of outstanding messages.
+    ///
+    /// The server will stop sending messages to a client when this many
+    /// messages are outstanding (i.e. that have not been acked or nacked).
+    ///
+    /// The server resumes sending messages when the outstanding message count
+    /// drops below this value.
+    ///
+    /// Use a value <= 0 to set no limit on the number of outstanding messages.
+    ///
+    /// # Example
+    ///
+    /// ```no_rust
+    /// # use google_cloud_pubsub::client::Subscriber;
+    /// # async fn sample() -> anyhow::Result<()> {
+    /// # let client = Subscriber::builder().build().await?;
+    /// let session = client.streaming_pull("projects/my-project/subscriptions/my-subscription")
+    ///     .set_max_outstanding_messages(100000)
+    ///     .start();
+    /// # Ok(()) }
+    /// ```
+    pub fn set_max_outstanding_messages<T: Into<i64>>(mut self, v: T) -> Self {
+        self.max_outstanding_messages = v.into();
+        self
+    }
+
+    /// Flow control settings for the maximum number of outstanding bytes.
+    ///
+    /// The server will stop sending messages to a client when this many bytes
+    /// of messages are outstanding (i.e. that have not been acked or nacked).
+    ///
+    /// The server resumes sending messages when the outstanding byte count
+    /// drops below this value.
+    ///
+    /// Use a value <= 0 to set no limit on the number of outstanding bytes.
+    ///
+    /// # Example
```

Comment:
nit: Same comment about default for this setting.

Reply:
Added